examples/win-proc-stat.c

```diff
@@ -0,0 +1,476 @@
+/* The LibVMI Library is an introspection library that simplifies access to
+ * memory in a target virtual machine or in a file containing a dump of
+ * a system's physical memory.  LibVMI is based on the XenAccess Library.
+ *
+ * Author: Sergey Kovalev (valor@list.ru)
+ *
+ * This file is part of LibVMI.
+ *
+ * LibVMI is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * LibVMI is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with LibVMI.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <glib.h>
+#include <signal.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <libvmi/libvmi.h>
+#include <libvmi/events.h>
+#include <json-c/json.h>
+
+/* Structure to pass data into event callback */
+struct cb_context {
+    addr_t* offsets;
+    GHashTable* stats;
+};
+
+/*
+ * Structures and functions to store stat data
+ */
+typedef union pid_tid {
+    struct {
+        uint64_t pid : 32;
+        uint64_t tid : 32;
+    };
+    uint64_t value;
+} pid_tid_t;
+
+guint pid_tid_hash(gconstpointer key)
+{
+    pid_tid_t k = { .value = (uint64_t)key };
+    uint32_t hash = (k.pid & 0xffff) + (k.tid << 16);
+
+    return hash;
+};
+
+int pid_tid_cmp(const void* a, const void* b)
+{
+    pid_tid_t a_ = { .value = (uint64_t)a };
+    pid_tid_t b_ = { .value = (uint64_t)b };
+
+    bool rc = 0;
+    if (a_.pid == b_.pid && a_.tid == b_.tid)
+        rc = 1;
+
+    return rc;
+}
+
+void print_stats(gpointer key, gpointer value, __attribute__((unused)) gpointer userdata)
+{
+    pid_tid_t k = { .value = (uint64_t)key };
+    uint64_t v = (uint64_t)value;
+
+    printf("PID:%d,TID:%d,COUNT:%ld\n", k.pid, k.tid, v);
+}
+
+/*
+ * Map offset enums to actual structure+member or global variable/function names.
+ */
+enum win_offsets {
+    KPCR_PRCB,
+    KPCR_PRCBDATA,
+    KPRCB_CURRENTTHREAD,
+    ETHREAD_CID,
+    CLIENT_ID_UNIQUETHREAD,
+    __WIN_OFFSETS_MAX
+};
+
+static const char* win_offset_names[__WIN_OFFSETS_MAX][2] = {
+    [KPCR_PRCB] = {"_KPCR", "Prcb" },
+    [KPCR_PRCBDATA] = {"_KPCR", "PrcbData" },
+    [KPRCB_CURRENTTHREAD] = { "_KPRCB", "CurrentThread" },
+    [ETHREAD_CID] = {"_ETHREAD", "Cid" },
+    [CLIENT_ID_UNIQUETHREAD] = {"_CLIENT_ID", "UniqueThread" },
+};
+
+static bool rekall_lookup_in_json(
```

Comment:
I don't think you should do this manually, there are already functions available in `libvmi_extra.h` that you can use to look up values in the json profiles, which works with either Rekall or Volatility profiles.